src/datetime.rs

```diff
@@ -210,26 +278,30 @@ impl crate::DateTime {
 
     /// Convert this [`DateTime`] to an RFC 3339 formatted string.
     pub fn to_rfc3339_string(self) -> String {
-        self.to_chrono()
-            .to_rfc3339_opts(chrono::SecondsFormat::AutoSi, true)
+        use std::result::Result;
+        match self.to_time().format(&Rfc3339) {
+            Result::Ok(s) => s,
+            Result::Err(e) => e.to_string(),
```

Comment:
I'm not thrilled about this but I couldn't think of a better option that wouldn't be a breaking change.

Reply:
From the source, the `time` [method](https://docs.rs/time/0.3.9/src/time/formatting/formattable.rs.html#158-236) fails if the year is outside [0,9999]; the chrono [equivalent](https://docs.rs/chrono/0.4.19/src/chrono/datetime.rs.html#436-476) doesn't have that check.

The `time` method has some other potential failure cases as well since it's a more general method, but for the specific purpose of formatting an `OffsetDateTime` I don't think they can happen.

Reply:
The `time` crate actually has a `large-dates` flag which expands this to +-99,999, which we may want to consider enabling here, as MongoDB technically supports any date that can be represented as an `i64` from the Unix epoch. Should we enable that flag, we'll want to let the maintainer know of our use case [here](https://github.com/time-rs/time/discussions/306) so that the feature flag stays around in the future.

Also, rather than erroring, it may be more appropriate to panic here. Could we test some inputs that error in `time` and see if they panic in `chrono`? If they do it might be okay for us to panic here as well. Another option would be to cap the date to `time`'s maximum value and use that for the date here. I think we do that in a few places already with `chrono`'s maximum date. If none of those work, then I agree we probably should deprecate this method and introduce a new one, since I think including error information as a String could be pretty cumbersome or dangerous.

Reply:
The `large-dates` flag is already used, actually (it's needed for the y10k comparisons to be meaningful), but it doesn't affect the range of dates supported by the rfc3339 format.  I went with the panic and deprecate option because outputting truncated values seems even more likely to cause bad downstream behavior than error strings.

Reply:
that solution SGTM. since we use the flag already it probably seems worth chiming in on that discussion linked above to let them know why

Reply:
And as a last check, are we okay with the risk of breaking user code with this? It's technically possible that a user has a super large date in their DB somewhere that used to format okay but will panic now via this method or fail to serialize via the serde helper.

Reply:
I think the risk is okay here - we're being loud about this change with the deprecation warning, and I think we should also call it out in the patch notes.  It's also worth considering that [the RFC itself](https://datatracker.ietf.org/doc/html/rfc3339) is explicit about only applying to years in the 0000-9999 range and tha the year field is fixed at four digits, so the chrono implementation is producing output that's probably not going to be accepted elsewhere.

Reply:
Oh nice, I didn't know that about the RFC, that definitely makes this change seem fine to me then.

Reply:
agree it seems fine 👍